app.py

```
@@ -28,6 +28,30 @@
 
+# Common printer communication mixin
+class PrinterCommunicationMixin:
+    def send_zpl_to_printer(self, printer_ip, printer_port, zpl_data):
+        try:
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+                sock.settimeout(10)  # Add timeout for better error handling
+                sock.connect((printer_ip, printer_port))
+                sock.sendall(zpl_data.encode('utf-8'))
+        except socket.timeout:
+            logging.error(f"Connection timeout to printer at {printer_ip}:{printer_port}")
+            raise Exception("Printer connection timeout")
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)